js/test/unit/ipc/helpers.ts

```diff
@@ -54,13 +54,13 @@ export abstract class ArrowIOTestHelper {
             await testFn(await this.writer(this.table).toUint8Array());
         };
     }
-    iterable(testFn: (iterable: Iterable<Uint8Array>) => void | Promise<void>) {
+    iterable(testFn: (iterable: Generator<Uint8Array>) => void | Promise<void>) {
         return async () => {
             expect.hasAssertions();
             await testFn(chunkedIterable(await this.writer(this.table).toUint8Array()));
         };
     }
-    asyncIterable(testFn: (asyncIterable: AsyncIterable<Uint8Array>) => void | Promise<void>) {
+    asyncIterable(testFn: (asyncIterable: AsyncGenerator<Uint8Array>) => void | Promise<void>) {
```

Comment:
This is a change in the TypeScript typings for generator functions. They didn't used to have a way to capture the error and return types of generator functions, but now they do. Most places we don't care about the error or return types, but have to use the Generator interfaces because TS 3.5+ requires it.

Reply:
Ah ok, thanks